language/java/language_test.go

```diff
@@ -249,3 +250,121 @@ func TestLanguageExecute(t *testing.T) {
 		})
 	})
 }
+
+func TestMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		RepositoryPath string
+
+		ExpectedMistakes []string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			temporaryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryPath, filepath.Base(tc.RepositoryPath))
+			require.NoError(t, osutil.CopyTree(tc.RepositoryPath, repositoryPath))
+
+			buffer, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Log(buffer.String())
+				}
+			}()
+
+			java := &Language{}
+			actualMistakes, actualErr := java.Mistakes(logger, repositoryPath)
+			require.NoError(t, actualErr)
+
+			for i, mistake := range actualMistakes {
+				index := strings.Index(mistake, "/src/main/java")
+				require.Greater(t, index, -1)
+				actualMistakes[i] = mistake[index:]
```

Comment:
Could you please add a comment that Maven prints paths with slashes even on Windows, so to split off the temporary path we need to be creative?